### PR TITLE
Runtime moduli

### PR DESCRIPTION
**Pull request by jellevos**
This PR extends #130. It adds residues for moduli decided at run-time.

### CODE REVIEW

File: src/uint/modular/mod.rs [hunk 1]
```diff
@@ -1,17 +1,62 @@
-use crate::UInt;
```
Comment (tarcieri) [inline src/uint/modular/mod.rs:1]: This PR looks good aside from the use of the 2015 edition module style (i.e. `mod.rs`)

We've adopted the 2018 edition module style, where this file would be modular.rs.

Unfortunately I don't think they ever added a lint for mixed module styles.

**Comment by tarcieri** [comment]
Will fix it up retroactively

File: src/uint/modular/mod.rs (hunk 1 above)
Comment (daxpedda) [reply src/uint/modular/mod.rs:1]: Is this what you mean?:
https://rust-lang.github.io/rust-clippy/master/index.html#mod_module_files

Comment (tarcieri) [reply src/uint/modular/mod.rs:1]: Oh ho, went looking for that the other day and didn't find it... weird

**Pull request merged**